datadotworld/client.py

```diff
@@ -41,52 +47,344 @@ def __init__(self, token=None, propsfile="~/.data.world",
             raise RuntimeError((
                 'you must either provide an API token to this constructor, or create a '
                 '.data.world file in your home directory with your API token'))
+
         self.protocol = protocol
-        self.queryHost = queryHost
+        self.query_host = query_host
+        self.apiHost = api_host
+
+        self._api_client = ApiClient(host="{}://{}/v0".format(protocol, api_host), header_name='Authorization',
+                                     header_value='Bearer {}'.format(token))
+        self._datasets_api = DatasetsApi(self._api_client)
+        self._uploads_api = UploadsApi(self._api_client)
+
+    # Dataset Operations
+
+    def get_dataset(self, dataset_key=None):
+        """Retrieve an existing dataset
+
+        Parameters
+        ----------
+        dataset_key : str
+            Dataset identifier, in the form of owner/id
+
+        Returns
+        -------
+        DatasetSummaryResponse
+            The dataset object
+
+        Raises
+        ------
+        ApiException
+            If a server error occurs
+
+        Examples
+        --------
+        >>> intro_dataset = dw.get_dataset('jonloyens/an-intro-to-dataworld-dataset')
+        >>> print(intro_dataset.description)
+        A dataset that serves as a quick introduction to data.world and some of our capabilities.  Follow along in \
+        the summary!
+        """
+        owner_id, dataset_id = dataset_key.split('/')
+        return self._datasets_api.get_dataset(owner_id, dataset_id)
+
+    def create_dataset(self, owner_id=None, dataset=None):
+        """Create a new dataset
+
+        Parameters
+        ----------
+        owner_id : str
+            Username of the owner of the new dataset
+        dataset : DatasetCreateRequest
```

Comment:
There are subtle differences right now, that justify different classes. For example, POST includes title but PUT doesn't. Visibility is required in PUT/POST, but not in PATCH. We'll leave it this way and may revisit in a future major release.